requests/adapters.py

```diff
@@ -284,7 +303,14 @@ def request_url(self, request, proxies):
         """
         proxy = select_proxy(request.url, proxies)
         scheme = urlparse(request.url).scheme
-        if proxy and scheme != 'https':
+        proxy_scheme = ''
+        if proxy:
+            proxy_scheme = urlparse(proxy).scheme
+
+        if proxy and proxy_scheme.lower().startswith('socks'):
```

Comment:
Ok, I'm going with that. I degolfed this a little bit to aid clarity, but not much.